packages/native/src/useLinking.tsx

```diff
@@ -296,12 +296,14 @@ export function useLinking(
             focusedRoute.name === route.name &&
             isEqual(focusedRoute.params, route.params)
           ) {
-            return route.path;
+            return route.path + location.hash;
           }
         }
       }
+      const path = getPathFromStateRef.current(state, configRef.current);
+      const isSameScreen = path.split('?')[0] === route?.path?.split('?')?.[0];
```

Comment:
`path` being the same doesn't always mean it's the same screen. the same screen could have 2 different paths if the params have changed for params in path segment, e.g. `/users/:userId`, 

To know if it's the same screen, you'd need to get the key of the current route (`route.key`) and compare it with the previous key (`route.key`).